private/buf/bufgen/config.go

```diff
@@ -163,36 +163,67 @@ func validateExternalConfigV1(externalConfig ExternalConfigV1, id string) error
 		return fmt.Errorf("%s: no plugins set", id)
 	}
 	for _, plugin := range externalConfig.Plugins {
-		if plugin.Name == "" && plugin.Remote == "" {
-			return fmt.Errorf("%s: plugin name or remote plugin name is required", id)
+		numPluginIdentifiers := 0
+		var pluginIdentifier string
+		for _, possibleIdentifier := range []string{plugin.Plugin, plugin.Name, plugin.Remote} {
+			if possibleIdentifier != "" {
+				numPluginIdentifiers++
+				// Doesn't matter if we reassign here - we only allow one to be set below
+				pluginIdentifier = possibleIdentifier
+			}
+		}
+		if numPluginIdentifiers == 0 {
+			return fmt.Errorf("%s: one of plugin, name or remote is required", id)
 		}
-		if plugin.Name != "" && plugin.Remote != "" {
-			return fmt.Errorf("%s: only one of plugin name and remote plugin name can be set", id)
+		if numPluginIdentifiers > 1 {
+			return fmt.Errorf("%s: only one of plugin, name, or remote can be set", id)
 		}
 		if plugin.Out == "" {
-			return fmt.Errorf("%s: plugin %s out is required", id, plugin.Name)
+			return fmt.Errorf("%s: plugin %s out is required", id, pluginIdentifier)
 		}
-		if plugin.Remote != "" {
-			if _, _, _, _, err := bufremoteplugin.ParsePluginVersionPath(plugin.Remote); err != nil {
+		if plugin.Plugin != "" {
+			if _, err := bufpluginref.PluginReferenceForString(pluginIdentifier); err == nil {
+				// plugin.Plugin is a remote plugin
+				if err := checkPathAndStrategyUnset(id, plugin, pluginIdentifier); err != nil {
+					return err
+				}
+			} else {
+				// plugin.Plugin is a local plugin - verify it isn't using an alpha remote plugin path
+				if _, _, _, _, err := bufremoteplugin.ParsePluginVersionPath(pluginIdentifier); err == nil {
+					return fmt.Errorf("%s: invalid local plugin", id)
+				}
+			}
+		} else if plugin.Remote != "" {
```

Comment:
We often try to prevent using `else` if we don't have to - we can reduce nesting a bit by using `continue` in each statement since each of the `plugin.{Plugin,Remote,Name}` are mutually exclusive at this point.

For example,

```go
if plugin.Plugin != "" {
  ...
  continue
}
if plugin.Remote {
  ...
  continue
}
...
```

With that said, this is actually a better opportunity to use a `switch` (due to the mutual exclusivity):

```go
switch {
  case plugin.Plugin != "":
    ...
  case plugin.Remote != "":
    ...
  case plugin.Name != "":
   ...
  default:
    // Unreachable, but handle error anyway.
}
```